### PR TITLE
[Model] Add Qwen3.5 sequence classification support

### PR DESCRIPTION
**Pull request by Daboolu**
## Motivation

SGLang supports Qwen3.5 generation and pooled embeddings, but it does not currently provide a native Qwen3.5 sequence-classification model implementation.

This is required by Qwen3.5 reward models whose checkpoints contain a sequence-level score head, for example:

```text
score.weight: [1, 2560]
problem_type: regression
num_labels: 1
```

The expected reward inference path is:

```text
Qwen3.5 backbone
→ raw LAST-token hidden state
→ score head
→ scalar reward
```

The existing Qwen3.5 embedding path applies L2 normalization to the pooled hidden state. While this is appropriate for embedding models, applying a reward head to that normalized representation changes the reward scale.

For sequence classification and reward inference, the score head must instead operate on the raw pooled hidden state.

This PR adds a dedicated `Qwen3_5ForSequenceClassification` implementation without changing existing Qwen3.5 generation or embedding behavior.

### Model implementation

- Add native `Qwen3_5ForSequenceClassification`.
- Reuse the existing multimodal Qwen3.5 backbone.
- Add a bias-free classification head:

  ```text
  Linear(text_hidden_size, num_labels, bias=False)
  ```

- Pool the raw LAST-token hidden state with:

  ```text
  PoolingType.LAST
  normalize=False
  ```

- Apply the score head through the existing `score_and_pool` infrastructure.
- Return raw logits/rewards without applying softmax.
- Skip loading generation-only `lm_head` weights for classification checkpoints.

### Shared Qwen3-VL integration

- Add an optional `create_lm_head` constructor argument while preserving `True` as the default.
- Add an overridable `_pool_hidden_states` hook to the shared multimodal forward path.
- Keep the existing normalized embedding pooler as the default implementation.
- Let the sequence-classification subclass override only the task-specific pooling/scoring step.
- Avoid constructing an unused vocabulary-sized generation head for classification models.

The defaults preserve existing Qwen3-VL and Qwen3.5 generation behavior.

### Runtime and model registration

Register `Qwen3_5ForSequenceClassification` in the existing Qwen3.5 capability paths:

- model registry
- classification model specification
- Qwen3.5 architecture detection
- non-generation model detection
- multimodal model support
- multimodal processor routing
- hybrid/Mamba runtime overrides
- Mamba radix-cache support
- CUDA graph capability lists
- multimodal scheduling
- shared-expert fusion gating

The classification model is exposed through SGLang's existing pooled-output serving mode.

### Tests

Add unit coverage for:

- native model-registry resolution
- classification task and execution-mode resolution
- preservation of multimodal capability
- LAST-token pooling
- use of the raw hidden state rather than an L2-normalized embedding
- score-head output
- shared-expert fusion gating

### Static checks

The following checks passed:

```bash
python -m compileall -q \
  python/sglang/srt/models/qwen3_5.py \
  python/sglang/srt/models/qwen3_vl.py \
  test/registered/unit/models/test_qwen3_5_sequence_classification.py

git diff --check
```

### Unit tests

Executed on an NVIDIA H800 environment:

```bash
python -m pytest -q \
  test/registered/unit/configs/test_embedding_model_spec.py \
  test/registered/unit/models/test_qwen3_5_sequence_classification.py \
  test/registered/unit/models/test_shared_experts_fusion_gates.py
```

Result:

```text
38 passed, 5 warnings, 2 subtests passed
```

The warnings were dependency deprecation/configuration warnings and did not indicate test failures.

### End-to-end reward validation

The implementation was validated with a BF16 Qwen3.5 4B regression reward checkpoint trained with ms-swift.

Relevant checkpoint properties:

```text
hidden_size: 2560
score.weight: [1, 2560]
problem_type: regression
num_labels: 1
```

The validation checkpoint stores:

```text
architectures = ["Qwen3_5ForConditionalGeneration"]
```

despite containing a regression `score.weight` head. For validation, the architecture was therefore explicitly overridden:

```bash
--json-model-override-args \
'{"architectures":["Qwen3_5ForSequenceClassification"]}'
```

This PR intentionally does not automatically reinterpret ordinary `Qwen3_5ForConditionalGeneration` checkpoints as reward models, because doing so could change normal generation behavior.

The exact 18-token input used for the SGLang/ms-swift comparison was:

```text
[
  248045, 846, 198, 9419, 11, 18553, 5941, 25899, 13,
  248046, 198, 248045, 74455, 198, 248068, 271, 248069, 271
]
```

Results:

```text
Input tokens:     18
SGLang reward:   -1.4296875
ms-swift reward: -1.4296875
Absolute error:   0.0
```

A separate comparison with Hugging Face used the same six raw-text token IDs:

```text
SGLang reward:    -1.484375
Hugging Face:     -1.4921875
Absolute error:    0.0078125
```

The remaining difference is one BF16 quantization step and is within the expected BF16 tolerance.

### Template alignment

The initial difference between SGLang and ms-swift raw-string requests was traced to preprocessing rather than model computation.

ms-swift converts a raw input string into a user message and applies the Qwen3.5 chat template with:

```text
add_generation_prompt=True
enable_thinking=False
```

This produces 18 tokens, while the plain SGLang string request contains six tokens.

When SGLang receives the same 18 token IDs, the reward matches ms-swift exactly:

```text
-1.4296875
```

This PR does not globally add ms-swift-specific template processing to `/v1/embeddings`. Callers remain responsible for applying preprocessing consistent with model training.

### Validation limitations

- Text reward inference was validated end to end.
- The multimodal model and processor paths are registered, but image/video reward inputs have not yet been validated end to end.
- The ms-swift validation checkpoint uses non-standard architecture metadata, so it requires the explicit architecture override shown above.

## Speed Tests and Profiling

No formal speed benchmark was collected for this PR.

This change does not introduce new CUDA kernels, attention implementations, or generation hot-path changes. It reuses the existing Qwen3.5 multimodal backbone and pooled-output infrastructure.

For sequence-classification requests, the generation `lm_head` is not constructed. The only task-specific operation added after pooling is a small linear score head.

The H800 end-to-end validation used Triton attention and linear-attention backends because the validation environment retained an older preinstalled SGLang kernel stack. Performance with the current fully matched main-branch dependency stack remains to be benchmarked.

## Checklist

- [x] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [x] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [x] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations). (N/A: this PR uses the existing pooled-output API and adds no new public API or CLI option.)
- [x] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed). (Accuracy results are provided above; speed benchmarking is not applicable to a new-kernel comparison because this PR adds no new kernel or attention implementation.)
- [x] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review and Merge Process

1. Ping Merge Oncalls to start the process. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - Common commands include `/tag-and-rerun-ci`, `/tag-run-ci-label`, `/rerun-failed-ci`
4. After green CI and required approvals, ask Merge Oncalls or people with Write permission to merge the PR.

---
### CI States

Latest PR Test (Base): :x: [Run #33238211852](https://github.com/sgl-project/sglang/actions/runs/33238211852)
Latest PR Test (Extra): :x: [Run #33238211707](https://github.com/sgl-project/sglang/actions/runs/33238211707)
Latest PR Test (AMD ROCm 7.2): :x: [Run #33238211802](https://github.com/sgl-project/sglang/actions/runs/33238211802)

### CODE REVIEW

**Comment by Daboolu** [comment]
Hi maintainers, could someone with CI permission please run `/tag-and-rerun-ci` for this PR?

Local unit tests and H800 end-to-end accuracy results are included in the PR description. Thank you!
